src/UCTSearch.cpp

```diff
@@ -239,24 +254,29 @@ int UCTSearch::get_best_move(passflag_t passflag) {
         }
     }
 
-    int visits = m_root.get_visits();
+    int visits = m_root->get_visits();
 
     // if we aren't passing, should we consider resigning?
     if (bestmove != FastBoard::PASS) {
         // resigning allowed
         if ((passflag & UCTSearch::NORESIGN) == 0) {
-            size_t movetresh = (m_rootstate.board.get_boardsize()
-                                * m_rootstate.board.get_boardsize()) / 4;
+            size_t movetresh = (m_rootstate->board.get_boardsize()
+                                * m_rootstate->board.get_boardsize()) / 4;
             // bad score and visited enough
             if (bestscore < ((float)cfg_resignpct / 100.0f)
                 && visits > 500
-                && m_rootstate.m_movenum > movetresh) {
+                && m_rootstate->m_movenum > movetresh) {
                 myprintf("Score looks bad. Resigning.\n");
                 bestmove = FastBoard::RESIGN;
             }
         }
     }
 
+    // TODO: Instead of searching for the child we could keep track of it.
+    // This is just an easy way to be safe.
```

Comment:
At this point you know you want the UCTNode out of m_children that has get_move() == bestmove, so all the object copying and hash lookups aren't needed?

Reply:
Created a new version of find_and_take_ownership in UCTNode to handle this one. Let me know if you're ok with overloading the same function name with different arguments.